src/orange/detail/OrientedBoundingZone.hh

```diff
@@ -0,0 +1,75 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/detail/OrientedBoundingZone.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "orange/OrangeData.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Oriented bounding zone for saftey distance calculations.
+ *
+ * Here, the bounding zone is defined by an inner and outer bounding box
+ * (specified via half-widths) transformed by the same transformation;
+ * pressumably the transformation of the volume they correspond to. The
+ * safety distance can be approximated as the minimum distance to the inner or
+ * outer box. For points lying between the inner and outer boxes, the safety
+ * distance is zero.
+ */
+class OrientedBoundingZone
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using Storage = NativeCRef<OrientedBoundingZoneData>;
+    //!@}
+
+  public:
+    // Construct from inner/outer half-widths and a corresponding transform
+    OrientedBoundingZone(Real3Id inner_hw_id,
+                         Real3Id outer_hw_id,
+                         TransformId transform_id,
+                         Storage* storage);
+
+    // Calculate the safety distance for any position inside the outer box
+    real_type safety_distance_inside(Real3 pos);
+
+    // Calculate the safety distance for any position outside the inner box
+    real_type safety_distance_outside(Real3 pos);
+
+    // Determine if a position is inside the inner box
+    bool is_inside_inner(Real3 const& pos);
+
+    // Determine if a position is inside the outer box
+    bool is_inside_outer(Real3 const& pos);
+
```

Comment:
To avoid having to repeat the transform on the point, I think we should have a single `SignedSense calc_sense(Real3 const& pos)`  that returns whether it's in/out/on. We should be able to use this during initialization and boundary crossings to find out whether we need to test the target volume.

Reply:
Between inner and outer box would come back as `SignedSense::on`. Is that close enough semantically?

Reply:
Yep, exactly. The bounding zone is like having a surface that's (often much) thicker than 1 ULP.